### PR TITLE
CRM-20145 line_item $0 entity_financial_trxn fix

### PR DESCRIPTION
**Pull request by lcdservices**
* [CRM-20145: line_item with $0 amount may error when creating civicrm_entity_financial_trxn](https://issues.civicrm.org/jira/browse/CRM-20145)

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -5418,7 +5418,8 @@ public static function getSalesTaxFinancialAccounts() {
   public static function createProportionalEntry($entityParams, $eftParams) {
     $paid = $entityParams['line_item_amount'] * ($entityParams['trxn_total_amount'] / $entityParams['contribution_total_amount']);
     // Record Entity Financial Trxn
-    $eftParams['amount'] = round($paid, 2);
+    // CRM-20145
+    $eftParams['amount'] = number_format((float)round($paid, 2), 2, '.', '');
```
Comment (JoeMurray) [inline CRM/Contribute/BAO/Contribution.php]: @lcdservices could we make this into a tiny utility function that can be used wherever we are about to store a money value in db? Currently we should always be storing 2 digits after the decimal in core, but cryptocurrencies will need more. @pradpnayak could you ensure we create/use the same utility function throughout work on sales taxes?

Comment (lcdservices) [inline CRM/Contribute/BAO/Contribution.php]: I've updated the PR accordingly.

**Comment by KarinG** [comment]
Actually - @eileenmcnaughton sorry... I've not been here yet... To be frank: I'm not sure what proportional  (as opposed to non-proportional ones?) financial transactions represent in terms of sales taxes - or which kind is needed when? Eh... I find eftParams very confusing - EFT in financial language is always used for Electronic Funds Transfer - and pretty sure that's not what's going on here. However if @lcdservices says this (whatever this is) is working for him if with this edit - then let's just get this in for him? Ok - I've got to get back to my own corporate Sales Tax return - due tomorrow - for real :-)

**Comment by eileenmcnaughton** [comment]
Well I think that is a blessing from Karin (of sorts) - still need the style issues resolved

**Comment by lcdservices** [comment]
@eileenmcnaughton style issues are fixed.
FYI, this PR doesn't really change anything about the proportional entry process aside from making sure the amount is formatted correctly.

**Pull request merged**